rfcs/text/0000-analytics-instrumentation-plan.md

```diff
@@ -0,0 +1,145 @@
+* Gather anonymized user browsing activity in the data portal and data browser
+* Gather anonymized data download events initiated from any user interface (web, CLI, API)
+* Gather anonymized data upload events
+* Enable restricted access control to the user metrics data set
+* Enable non-developers to access and analyze the resulting data set to improve the HCA DCP inform the HCA scientific
+  leadership team
+
+### User Stories or Use Cases
+
+* As an [HCA program, product or project manager](https://github.com/HumanCellAtlas/dcp-community/blob/master/charters/PM/charter.md)
+  (PM), I want to analyze quantitative metrics of user actions on DCP, to make informed decisions on product direction.
+* As an HCA UX researcher, I want to analyze quantitative metrics of user actions on DCP, to draw insights on our users'
+  needs and experiences using the DCP.
+
+## Data Privacy
+
+This plan has been reviewed and approved by the HCA GDPR Task Force.
+
+To make the plan GDPR compliant, an amended privacy policy will be deployed to the data portal when the analytics framework ships.
+
+## Detailed Design
+
+We’ve selected Google Analytics (GA) as our tracking solution. We briefly considered alternatives, like building our own
+and other commercial products. We chose Google Analytics for the following reasons:
+
+* **Community precedent** - Multiple comparable scientific data platforms like ENCODE presently use Google Analytics.
+  This is easy to verify by visiting their landing pages and checking for the Google Analytics javascript tag.
+* **Usability** - Google Analytics is widely used by non-developers, allowing point-and-click access.
+* **Cost** - Google Analytics is free for the user activity levels we anticipate (<10M events per month).
+* **Privacy policy** - Google Analytics has a strong privacy policy that prevents Google from accessing and/or
+  aggregating user activity data.
+* **Access controls** - Google Analytics has a robust access control system that makes it easy to extend and manage
+  access. This includes transferring ownership of a tracking account to a different admin.
+
+Downsides also exist. A primary one is that GA is built primarily for traditional web publishers and e-commerce. HCA DCP
+does not fit this mold. Thus many core GA features will be distractions, and features like data schema flexibility will
+be lacking. These downsides were not significant enough to warrant not using GA.
+
+### Configuration
+
+1. We configure a [tracking ID](https://support.google.com/analytics/answer/1008080?hl=en) in the GA console for each of
+   the deployment environments
+    1. `dev` - for local development and tests on the dev deployment
+    2. `integration` - for integration tests
+    3. `staging` - for the staging environment
+    4. `prod` - for the production deployment and real usage in the browser
+
+Dividing up metrics collection by deployment environment allows us to do the following. * test that metrics are flowing
```

Comment:
Are the asterisks intended for formatting? They are just showing up as is when viewing the file.

Reply:
They were a list that got mangled. Fixed.